lib/spack/spack/binary_distribution.py

```diff
@@ -1694,22 +1696,22 @@ def relocate_package(spec, allow_root):
     old_rel_prefix = buildinfo.get("relative_prefix")
     old_prefix = os.path.join(old_layout_root, old_rel_prefix)
     rel = buildinfo.get("relative_rpaths")
-    prefix_to_hash = buildinfo.get("prefix_to_hash", None)
-    if old_rel_prefix != new_rel_prefix and not prefix_to_hash:
+
+    # In the past prefix_to_hash was the default and externals were not dropped, so prefixes
+    # were not unique.
+    if "hash_to_prefix" in buildinfo:
+        hash_to_old_prefix = buildinfo["hash_to_prefix"]
+    elif "prefix_to_hash" in buildinfo:
+        hash_to_old_prefix = dict((v, k) for (k, v) in buildinfo["prefix_to_hash"].items())
```

Comment:
Should we emit a warning and say that this version of the buildcache will not be supported in Spack v0.21 or v0.22?

Reply:
Let's add a warning from Spack 0.20

Reply:
Okay, I'll open an issue to remind us of that.